packages/eslint-plugin/docs/rules/no-unnecessary-type-parameters.mdx

```diff
@@ -87,6 +87,111 @@ This is because the type parameter `T` relates multiple methods in the `T[]` tog
 Therefore, this rule won't report on type parameters used as a type argument.
 That includes type arguments given to global types such as `Array` (including the `T[]` shorthand and in tuples), `Map`, and `Set`.
 
+## FAQ
+
+### I'm using the type parameter inside the function, so why is the rule reporting?
+
+You might be surprised to that the rule reports on a function like this:
+
+```ts
+function log<T extends string>(string1: T): void {
+  const string2: T = string1;
+  console.log(string2);
+}
+```
+
+After all, the type parameter `T` relates the input `string1` and the local variable `string2`, right?
+However, this usage is unnecessary, since we can achieve the same results by replacing all usages of the type parameter with its constraint.
+That is to say, the function can always be rewritten as:
+
+```ts
+function log(string1: string): void {
+  const string2: string = string1;
+  console.log(string2);
+}
+```
+
+Therefore, this rule only counts usages of a type parameter in the _signature_ of a function, method, or class, but not in the implementation. See also [#9735](https://github.com/typescript-eslint/typescript-eslint/issues/9735)
```

Comment:
I'm guessing users will be confused by implicit return types:

```ts
// "wait, this is only used in the body, not in the signature?"
function identity<T>(value: T) {
  return value;
}
```

Do you think it would be useful to mention those?

Reply:
good idea. added an entry